model/lib/extensions/session_extension.dart

```diff
@@ -2,22 +2,34 @@
 import 'dart:ui';
 
 import 'package:core/presentation/extensions/uri_extension.dart';
+import 'package:get/get_utils/src/extensions/string_extensions.dart';
 import 'package:jmap_dart_client/jmap/account_id.dart';
 import 'package:jmap_dart_client/jmap/core/capability/calendar_event_capability.dart';
 import 'package:jmap_dart_client/jmap/core/capability/capability_identifier.dart';
 import 'package:jmap_dart_client/jmap/core/capability/capability_properties.dart';
+import 'package:jmap_dart_client/jmap/core/capability/default_capability.dart';
 import 'package:jmap_dart_client/jmap/core/capability/empty_capability.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
 import 'package:jmap_dart_client/jmap/core/session/session.dart';
 import 'package:model/error_type_handler/account_exception.dart';
+import 'package:model/error_type_handler/unknown_address_exception.dart';
+import 'package:model/error_type_handler/unknown_uri_exception.dart';
 import 'package:model/model.dart';
+import 'package:model/principals/capability_principals.dart';
 import 'package:uri/uri.dart';
 
 extension SessionExtension on Session {
 
   String getDownloadUrl({String? jmapUrl}) {
-    final downloadUrlValid = jmapUrl != null
-      ? downloadUrl.toQualifiedUrl(baseUrl: Uri.parse(jmapUrl))
-      : downloadUrl;
+    final Uri downloadUrlValid;
+    if (jmapUrl != null) {
+      downloadUrlValid = downloadUrl.toQualifiedUrl(baseUrl: Uri.parse(jmapUrl));
+    } else if (downloadUrl.hasOrigin) {
+      downloadUrlValid = downloadUrl;
+    } else {
+      throw UnknownUriException();
```

Comment:
this function was use in many places without error catching, so IMO, please use `tryParse` instead